infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -416,7 +419,7 @@ private boolean clusterPresentInTopology(TopologyDescriptor topology, boolean up
      * Retrieve local Corfu Endpoint
      */
     private String getCorfuEndpoint(String localEndpoint, int corfuPort) {
-        return NodeLocator.parseString(localEndpoint).getHost() + ":" + corfuPort;
+        return getVersionFormattedEndpointURL(localEndpoint, corfuPort);
```

Comment:
local endpoint contains the whole address with the port - localhost:9010.  Dont we have to extract the address part before invoking getVersionFormattedEndpointURL?

Reply:
Done, Thank you.

Reply:
I had to revert.
Usage of `getCorfuEndpoint` -
```
localCorfuEndpoint = getCorfuEndpoint(getLocalHost(), localClusterDescriptor.getCorfuPort());
..., and `getLocalHost()` declaration -
 
 private String getLocalHost() {
     return NodeLocator.parseString(
         getVersionFormattedHostAddress(serverContext.getLocalEndpoint())
     ).getHost();            // <----- returning only the host.
}
```

Usage in an if condition:
`if (getLocalHost().equals(DefaultClusterConfig.getDefaultHost())) {`, 
and `DefaultClusterConfig.getDefaultHost()` is always `localhost`

So port was not involved in the `localEndpoint`. I have renamed variables accordingly and made `getLocalHost()` a bit simpler now. It was a bit ambiguous.